src/server/scripts/EasternKingdoms/SunkenTemple/instance_sunken_temple.cpp

```diff
@@ -189,20 +325,156 @@ class instance_sunken_temple : public InstanceMapScript
+                        }
+                    Events.Reset();
+                    SetBossState(DATA_AVATAR_OF_HAKKAR, NOT_STARTED);
+                }
+            }
+            break;
+            default:
+                break;
+            }
+            return true;
+        }
+
+        void SetData(uint32 type, uint32 data) override
+        {
+            switch (type)
+            {
+            case EVENT_STATE:
                 State = data;
-         }
+                break;
+            case DATA_ETERNAL_FLAME:
+                CountFlames = data;
+                if (Creature* shade = instance->GetCreature(ShadeOfHakkarGUID)) {
+                    switch (CountFlames)
+                    {
+                    case 1:
+                        shade->AI()->Talk(SAY_ETHERNAL_FLAME_SHADE1);
+                        break;
+                    case 2:
+                        shade->AI()->Talk(SAY_ETHERNAL_FLAME_SHADE2);
+                        break;
+                    case 3:
+                        shade->AI()->Talk(SAY_ETHERNAL_FLAME_SHADE3);
+                        break;
+                    case 4:
+                        shade->DespawnOrUnsummon();
+                        if (Creature* avatar = instance->GetCreature(AvatarOfHakkarGUID))
+                        {
+                            avatar->AI()->DoAction(ACTION_FIGHT_STATE_AVATAR);
+                            avatar->AI()->DoZoneInCombat();
+                        }
+                        Events.Reset();
+                        break;
+                    default:
+                        break;
+                    }
+                }
+                break;
+            default:
+                break;
+            }
+        }
 
-         uint32 GetData(uint32 type) const override
-         {
-            if (type == EVENT_STATE)
+        uint32 GetData(uint32 type) const override
+        {
+            switch (type)
+            {
+            case EVENT_STATE:
                 return State;
+                break;
+            case DATA_AVATAR_OF_HAKKAR:
+                return AvatarOfHakkarEventState;
+                break;
+            case DATA_ETERNAL_FLAME:
+                return CountFlames;
+                break;
+            default:
+                break;
+            }
             return 0;
-         }
-    };
+        }
 
+        ObjectGuid GetGuidData(uint32 type) const override
+        {
+            switch (type)
+            {
+            case DATA_AVATAR_OF_HAKKAR:
+                return AvatarOfHakkarGUID;
+                break;
+            default:
+                break;
+            }
+            return ObjectGuid::Empty;
+        }
+
+        /*void ReadSaveDataMore(std::istringstream& data) override
+        {
+            data >> AvatarOfHakkarEventState;
+        }
+
+        void WriteSaveDataMore(std::ostringstream& data) override
+        {
+            data << AvatarOfHakkarEventState;
+        }*/
```

Comment:
Do you need this code for later, maybe as to ask one of the TC members to validate the code or similar? Unless we are going to activate the code, it is recommended not to add commented-out code.